### ISSUE TITLE
`Hazard.from_xarray_raster` requires that "event" dimension can be interpreted as datetime

### ISSUE THREAD

**Issue by peanutfun** [CLIMADA-project/climada_python#829]
https://github.com/CLIMADA-project/climada_python/pull/795 has an unexpected repercussion (for me): The inserted hazard data now has to provide an "event" dimension that can be interpreted as datetime. I think this is unnecessarily strict. Consider the following code:

```python
import xarray as xr
import numpy as np
from climada.hazard.base import Hazard

intensity = np.array([[[0, 1, 2], [3, 4, 5]], [[6, 7, 8], [9, 10, 11]]])
number = np.array([1, 2])
latitude = np.array([0, 1])
longitude = np.array([0, 1, 2])
dset = xr.Dataset(
    {
        "intensity": (["number", "latitude", "longitude"], intensity),
    },
    dict(number=number, latitude=latitude, longitude=longitude),
)

Hazard.from_xarray_raster(
    dset, "", "", intensity="intensity", coordinate_vars={"event": "number"}
)
```
This results in
```
TypeError: '.dt' accessor only available for DataArray with datetime64 timedelta64 dtype or for arrays containing cftime datetime objects.
```
Providing event names explicitly does **not** help, because the default values are always evaluated in the current implementation.

```python
dset["event_name"] = ["a", "b"]
Hazard.from_xarray_raster(
    dset, "", "", intensity="intensity", coordinate_vars={"event": "number"}
)  # Same error
```
Note that following line is *always* evaluated, even if the default value is not used: https://github.com/CLIMADA-project/climada_python/blob/0b06f3f8277b4c392a7e9cdd80a5c258c1ff1175/climada/hazard/base.py#L843

So, could we maybe just drop this default value altogehter? The default event name could just be a list of empty strings.

**Comment by chahank** [comment]
I think it would be a bad idea to have an empty string for the name.

**Comment by peanutfun** [comment]
Why? Do you have another suggestion? Currently, one has to manually add dates which is horribly cumbersome if no such information is provided in the dataset.

**Comment by chahank** [comment]
Because it is a property from the hazard from which one can select etc. Also some plotting functions may require it. 

But ultimately maybe you are right and it is not needed. To be tested.

**Comment by peanutfun** [comment]
Resolved by #837
